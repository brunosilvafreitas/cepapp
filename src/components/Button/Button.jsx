import React from 'react'
import styled from 'styled-components'

const StyledButton = styled.button`
  border-radius: 5px;
  color: #fff;
  background-color: #007bff;
  border-color: #007bff;
  box-shadow: none;
  box-sizing: border-box;
  padding: 6px 10px;
  cursor: pointer;

  &:disabled {
    opacity: 0.4;
    cursor: not-allowed;
  }
`

const Button = ({ children, ...other }) => (
  <StyledButton {...other}>
    {children}
  </StyledButton>
)

Button.displayName = 'Button'

export default Button
