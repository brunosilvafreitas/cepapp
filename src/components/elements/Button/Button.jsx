import React from 'react'
import styled from 'styled-components'

const StyledButton = styled.button`
  border-radius: 5px;
`

const Button = ({ children, ...other }) => (
  <StyledButton {...other}>
    {children}
  </StyledButton>
)

Button.displayName = 'Button'

export default Button
