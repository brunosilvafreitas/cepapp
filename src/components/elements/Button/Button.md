# Button

Componente botão padrão

## Aplicação

```
<Button>Clique aqui!</Button>
```