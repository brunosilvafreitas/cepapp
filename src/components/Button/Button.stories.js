import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, text } from '@storybook/addon-knobs'
import { withReadme } from 'storybook-readme'
import Button from './Button'
import docs from './Button.md'

storiesOf('@cepapp/components/Button', module)
  .addDecorator(withKnobs)
  .add('default', withReadme(docs, () => (
    <Button>{text('Texto do bot√£o', 'Clique aqui!')}</Button>
  )))
