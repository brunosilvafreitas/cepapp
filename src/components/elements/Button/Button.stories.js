import React from 'react'
import { storiesOf, addDecorator } from '@storybook/react'
import { withKnobs, text } from '@storybook/addon-knobs'
import { withReadme } from 'storybook-readme'
import Button from './Button'
import docs from './Button.md'

storiesOf('@cepapp/components/elements/Button', module)
  .addDecorator(withKnobs)
  .add('default', withReadme(docs, () => (
    <Button>{text('Texto do bot√£o', 'Clique aqui!')}</Button>
  )))
