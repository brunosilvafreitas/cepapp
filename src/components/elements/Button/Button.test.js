import React from 'react'
import { render } from 'react-testing-library'
import Button from '../Button'

describe('[Element] Button', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const { container } = render(<Button>Clique aqui!</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
