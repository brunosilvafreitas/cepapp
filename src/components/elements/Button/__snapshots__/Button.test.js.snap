// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Element] Button deve conseguir renderizar [snapshot] 1`] = `
.c0 {
  border-radius: 5px;
}

<button
  class="c0"
>
  Clique aqui!
</button>
`;
